### PR TITLE
initial pbft view change commit

### PR DESCRIPTION
**Pull request by chaosma**
* Add minimum view change structures and refactoring consensus codes
* Add timeout class for view change
* Not touch main code, large refactoring will come later
* Add README for PBFT including view change

### CODE REVIEW

File: consensus/view_change.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+// Mode return the current node mode
+func (pm *PbftMode) Mode() Mode {
+	return pm.mode
+}
+
+// SetMode set the node mode as required
+func (pm *PbftMode) SetMode(m Mode) {
+	pm.mux.Lock()
+	pm.mux.Unlock()
```
Comment (LeoHChen) [inline consensus/view_change.go]: defer

File: internal/utils/timer.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package utils
```
Comment (LeoHChen) [inline internal/utils/timer.go]: no test on this timer package.

File: consensus/view_change.go (hunk 1 above)
Comment (chaosma) [reply consensus/view_change.go]: done

File: internal/utils/timer.go (hunk 2 above)
Comment (chaosma) [reply internal/utils/timer.go]: done

**Pull request merged**